a2rchi/chains/utils/token_limiter.py

```diff
@@ -17,9 +20,18 @@ def reduce_tokens_below_limit(self, docs: list[Document]) -> list[Document]:
         if not docs:
             return []
 
-        tokens = [
-            self.llm.get_num_tokens(doc.page_content) for doc in docs
-        ]
+        tokens = []
+        for i, doc in enumerate(docs):
+            try:
+                doc_tokens = self.llm.get_num_tokens(doc.page_content)
+                if doc_tokens is None or doc_tokens < 0:
+                    doc_tokens = max(len(doc.page_content) // 4, 1)
```

Comment:
would be cleaner to error out here so as to go to the Exception, which would reduce code duplication